src/rpc/blockchain.cpp

```diff
@@ -1645,34 +1646,141 @@ UniValue MempoolInfoToJSON(const CTxMemPool& pool)
                         {RPCResult::Type::STR_AMOUNT, "minrelaytxfee", "Current minimum relay fee for transactions"},
-                        {RPCResult::Type::NUM, "unbroadcastcount", "Current number of transactions that haven't passed initial broadcast yet"}
+                        {RPCResult::Type::NUM, "unbroadcastcount", "Current number of transactions that haven't passed initial broadcast yet"},
+                        {RPCResult::Type::OBJ, "fee_histogram", "",
+                            {
+                                {RPCResult::Type::OBJ_DYN, "fee_rate_groups", "",
+                                {
+                                    {RPCResult::Type::OBJ, "<fee_rate_group>", "Fee rate group named by its lower bound (in " + CURRENCY_ATOM + "/vB), identical to the \"from\" field below",
+                                    {
+                                        {RPCResult::Type::NUM, "size", "Cumulative size of all transactions in the fee rate group"},
+                                        {RPCResult::Type::NUM, "count", "Number of transactions in the fee rate group"},
+                                        {RPCResult::Type::NUM, "fees", "Cumulative fees of all transactions in the fee rate group (in " + CURRENCY_ATOM + "/vB)"},
+                                        {RPCResult::Type::NUM, "from", "Group contains transactions with fee rates equal or greater than this value (in " + CURRENCY_ATOM + "/vB)"},
+                                        {RPCResult::Type::NUM, "to", "Group contains transactions with fee rates equal or less than this value (in " + CURRENCY_ATOM + "/vB)"},
+                                    }}}},
+                                {RPCResult::Type::NUM, "total_fees", "Total available fees in mempool (in " + CURRENCY_ATOM + "/vB)"},
+                            }},
                     }},
                 RPCExamples{
-                    HelpExampleCli("getmempoolinfo", "")
-            + HelpExampleRpc("getmempoolinfo", "")
+                    HelpExampleCli("getmempoolinfo", "") +
+                    HelpExampleCli("getmempoolinfo", R"("[0, 1, 2, 3, 4, 5, 6, 7, 8, 10, 12, 14, 17, 20, 25, 30, 40, 50, 60, 70, 80, 100, 120, 140, 170, 200]")") +
+                    HelpExampleRpc("getmempoolinfo", "") +
+                    HelpExampleRpc("getmempoolinfo", R"([0, 1, 2, 3, 4, 5, 6, 7, 8, 10, 12, 14, 17, 20, 25, 30, 40, 50, 60, 70, 80, 100, 120, 140, 170, 200])")
                 },
         [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
 {
-    return MempoolInfoToJSON(EnsureAnyMemPool(request.context));
+    MempoolHistogramFeeRates feelimits;
+    std::optional<MempoolHistogramFeeRates> feelimits_opt = std::nullopt;
+
+    if (!request.params[0].isNull()) {
+        const UniValue feelimits_univalue = request.params[0].get_array();
+
+        if (feelimits_univalue.size() == 0 || feelimits_univalue.size() > 30) {
```

Comment:
The idea behind the limit was that I didn't want to let it be unbounded (`MempoolInfoToJSON` acquires `pool.cs` lock). If this is not a concern, I can remove the check or I can increase it substantially. 

I'm not sure what option is better really.

Reply:
@0xB10C Would it make more sense to have a limit or having no limit? I'm not sure what is preferrable.

Reply:
Good point on the lock.. Can we benchmark the time the RPC takes with 10, 100. 1000, 10000 histogram bins? I think if it scales linear O(n) and somewhere between 10ms or so, then it's fine to have no limit. Users selecting a million bins should expect that this could take a bit longer than only 10 bins.

Reply:
https://github.com/bitcoin/bitcoin/pull/21422/files#diff-decae4be02fb8a47ab4557fe74a9cb853bdfa3ec0fa1b515c0a1e5de91f4ad0bR1685-R1699 takes `O(#mempoolTxs * #bins)` 

Looking at https://jochen-hoenicke.de/queue/#BTC,1y,count, to have some sense about numbers, it can be something like: 20_000 * 10_000 = 200_000_000 that should be computed in a less than a second.

If you can test it, it would be great. Thinking about it, maybe it's easy just to modify the test we have to add a lot of transactions and just measure the time. So it should be relatively easy I guess.